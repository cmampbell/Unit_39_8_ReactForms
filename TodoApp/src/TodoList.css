.TodoList-todos {
    display: inline-flex;
    flex-direction: column;
    align-items: center;
}