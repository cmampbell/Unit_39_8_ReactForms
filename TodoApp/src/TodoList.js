import React, { useState } from "react";
import { v4 as uuid } from "uuid"
import Todo from "./Todo"
import NewTodoForm from "./NewTodoForm"
import "./TodoList.css"

const TodoList = () => {
    const [todos, setTodos] = useState([])

    const addTodo = ( todo ) => {
        setTodos(todos =>
            [...todos, todo]
        )
    }

    const deleteTodo = (todoToDelete) => {
        setTodos(todos => 
            todos.filter(todos => todoToDelete !== todos)
            )
    }

    return (
        <div className="TodoList">
            <h1 className="TodoList-h1">To Do List</h1>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList-todos">
            {todos.map(todo => <Todo text={todo} key={uuid()} deleteTodo={deleteTodo}/>)}
            </div>
        </div>
    )
}

export default TodoList;