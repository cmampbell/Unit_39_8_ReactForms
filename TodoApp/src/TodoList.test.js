import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import TodoList from './TodoList';


it('renders without crashing', function() {
    render(<TodoList />);
});

it('matches snapshot', function() {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('adds todo on form submit', function() {
    const {getByLabelText, getByText} = render(<TodoList />);

    fireEvent.change(getByLabelText('Task:'), {target:{value:'Test'}})

    fireEvent.click(getByText('Add Todo'))

    expect(getByText('X')).toBeInTheDocument()
    expect(getByText('Test')).toBeInTheDocument()
})