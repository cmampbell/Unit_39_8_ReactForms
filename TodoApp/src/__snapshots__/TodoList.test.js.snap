// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1
      class="TodoList-h1"
    >
      To Do List
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoForm-label"
        for="todo"
      >
        Task:
      </label>
      <input
        class="NewTodoForm-input"
        id="todo"
        name="todo"
        placeholder="Type todo here"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-btn"
      >
        Add Todo
      </button>
    </form>
    <div
      class="TodoList-todos"
    />
  </div>
</DocumentFragment>
`;
